extras/parse_travis_log.py

```diff
@@ -49,81 +53,88 @@ def process_installcheck(f):
     and failed tests. Especially, it parses the output after
     'NEST Testsuite Summary'.
     """
-    res, line = process_until(f, 'NEST Testsuite Summary')
-    if res:
-        f.readline()  # -------
-        f.readline()  # NEST Executable: ...
-        f.readline()  # SLI Executable: ...
-        line = f.readline()  # Total number of tests: <int>
-        total = int(line.split(' ')[-1])
-        f.readline()  # Passed: <int>
-        line = f.readline()  # Failed: <int> (<int> PyNEST)
-        failed = [int(s) for s in line.split() if s.isdigit()][0]
-        return total, failed, line
+    in_test = False
+    in_results = False
+    total = None
+    failed = None
+    for line in open(f, "r"):
+        if line.strip() == "======= Test NEST start =======":
+            in_test = True
+            total = 0
+            failed = -1
+        elif line.strip() == "======= Test NEST end =======":
+            return total, failed
+        elif in_test and line.strip() == "NEST Testsuite Summary":
+            in_results = True
+        elif in_results and "Total number of tests:" in line:
+            total = int(line.split(' ')[-1])
+        elif in_results and "Failed" in line:
+            failed = [int(s) for s in line.split() if s.isdigit()][0]
+    return total, failed
 
 
 def process_changed_files(f):
     """
-    Run through the file f until a line starts with '+file_names='.
+    Run through the file f until a line starts with 'file_names='.
     The equal sign is followed by a space separated list of changed files
     in the PR. Returns that list.
     """
-    while True:
-        line = f.readline()
-        if not line:
-            return False, line
-
-        if line.startswith('+file_names='):
-            return (filter(lambda x: x != '',
-                           line.strip().split('=')[1].split(' ')), line)
-
-
-def process_vera(f, filename):
+    in_changed_files_section = False
+    for line in open(f, "r"):
+        if (not in_changed_files_section and
+                line.strip() == "======= Extract changed files start ======="):
+            in_changed_files_section = True
+        if in_changed_files_section and line.startswith('file_names='):
```

Comment:
Shouldn't this be `line.strip().startswith('file_names=')` to be on the safe side? Applies also to all other uses of `line.startswith()`.